Userland/Libraries/LibWeb/ImageDecoding.h

```diff
@@ -1,15 +1,40 @@
 /*
  * Copyright (c) 2020-2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, Dex♪ <dexes.ttp@gmail.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #pragma once
 
-#include <LibImageDecoderClient/Client.h>
+#include <AK/RefPtr.h>
+#include <AK/Vector.h>
+#include <LibGfx/Bitmap.h>
 
-namespace Web {
+namespace Web::ImageDecoding {
 
-ImageDecoderClient::Client& image_decoder_client();
+struct Frame {
+    RefPtr<Gfx::Bitmap> bitmap;
+    size_t duration { 0 };
+};
+
+struct DecodedImage {
+    bool is_animated { false };
+    u32 loop_count { 0 };
+    Vector<Frame> frames;
+};
+
+class Decoder : public RefCounted<Decoder> {
```

Comment:
If we're going to store this guy in a "singleton" instance that lives until the end of the process, then it really doesn't make much sense to have him be ref-counted. Is this needed for compatibility with other LibWeb interfaces?

Reply:
This is apparently needed for compatibility with the IPC protocol. Switching to an OwnPtr makes the compiler angry - something about the IPC decoder not being happy. I haven't really managed to investigate why that's the case, though, so there's probably a way to make it work by making the IPC decoder more lenient and/or moving things around in another way? Not sure.

Reply:
Fair enough. worth looking into, but not necessary right now.